File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/pojo/Bot.java
package cz.muni.fi.umimecesky.pojo;

import android.widget.ImageView;

import cz.muni.fi.umimecesky.logic.RobotLogic;
import cz.muni.fi.umimecesky.utils.Util;

public class Bot extends AbstractRobot {

    private final RobotLogic logic;

    public Bot(ImageView view, RobotLogic logic) {
        super(view, logic.hopsPerCorrect());
        this.logic = logic;
    }

    public void processBotMove() {
        boolean canMove = Util.probabilityTrue(getLogic().correctnessRatio());
        if (canMove) {
            applyCorrect();
        } else {
            applyWrong();
        }
    }


    public RobotLogic getLogic() {
        return logic;
    }

    @Override
    public void applyCorrect() {
        moveForward();
    }

    @Override
    public void applyWrong() {
        if (getLastWasCorrect()) { // bot moves back only once
            moveBackward();
        }
    }
}

File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/pojo/Bot.kt
package cz.muni.fi.umimecesky.pojo

import android.widget.ImageView
import cz.muni.fi.umimecesky.logic.RobotLogic
import cz.muni.fi.umimecesky.logic.RobotLogic.Companion.probabilityTrue

class Bot(view: ImageView, val logic: RobotLogic) : AbstractRobot(view, logic.hopsPerCorrect()) {

    fun processBotMove() {
        val canMove = probabilityTrue(logic.correctnessRatio())
        if (canMove) {
            applyCorrect()
        } else {
            applyWrong()
        }
    }

    override fun applyCorrect() = moveForward()

    override fun applyWrong() {
        // bot moves back only once
        if (lastWasCorrect) moveBackward()
    }
}
